crates/agglayer-clock/src/block.rs

```diff
@@ -4,19 +4,36 @@ use std::{
         atomic::{AtomicU64, Ordering},
         Arc,
     },
+    time::Duration,
 };
 
-use ethers::{
-    providers::{Middleware, PubsubClient},
-    types::Block,
+use alloy::{
+    network::Ethereum,
+    providers::{
+        fillers::{BlobGasFiller, ChainIdFiller, FillProvider, GasFiller, JoinFill, NonceFiller},
+        Identity, Provider, ProviderBuilder, RootProvider, WsConnect,
+    },
+    pubsub::{ConnectionHandle, PubSubConnect, PubSubFrontend},
+    rpc::client::ClientBuilder,
+    transports::{impl_future, TransportErrorKind, TransportResult},
 };
-use futures::StreamExt as _;
+use backoff::ExponentialBackoff;
 use tokio::sync::{broadcast, oneshot};
 use tokio_util::sync::CancellationToken;
 use tracing::{debug, error, info, trace, warn};
 
 use crate::{Clock, ClockRef, Error, Event, BROADCAST_CHANNEL_SIZE};
 
+type BlockProvider = FillProvider<
+    JoinFill<
+        Identity,
+        JoinFill<GasFiller, JoinFill<BlobGasFiller, JoinFill<NonceFiller, ChainIdFiller>>>,
+    >,
```

Comment:
Nitpick:

```suggestion
    JoinFill<
        Identity,
        Ethereum::RecommendedFillers,
    >,
```

(probably needs to import / qualify with the `RecommendedFillers` trait, will not work as suggested)

Reply:
I tried to use that format earlier, but it doesn't reduce the boilerplate as I needed to define `Ethereum as ...`